stat/spatial/spatial.go

```diff
@@ -43,11 +42,21 @@ func GetisOrdGStar(i int, data, weights []float64, locality mat.Matrix) float64
 	n := float64(len(data))
 	mean, std := stat.MeanStdDev(data, weights)
 	var dwd, dww, sw float64
-	for j, v := range data {
-		w := locality.At(i, j)
-		sw += w
-		dwd += w * v
-		dww += w * w
+	if band, ok := locality.(mat.Banded); ok {
```

Comment:
I've tried this, and it's a little more complicated than I thought it would be. You really need a `Do`, a `DoRow` and a `DoCol`, or a way to restrict to columns/rows. The `Do` method works find for GetisOrd, though it needs to return without work for m-1 rows.